### PR TITLE
Fix Xcode 11 build

### PR DESCRIPTION
**Pull request by ericlewis**
## Summary

Fixes build in Xcode 11 beta, the signature for `__unused` was changed. This adds a new check for the new style.

## Changelog

[iOS] [Fixed] - Xcode 11 beta build

## Test Plan

Build & run RNTester

### CODE REVIEW

**Comment by ericlewis** [comment]
Failing tests seem unrelated.

**Comment by ericlewis** [comment]
props to @zhongwuzw for the debug assist!

**Comment by zhongwuzw** [comment]
@ericlewis 💯 `NSArray` related crash is also resolved?

**Comment by Bijiabo** [comment]
Thank you!

**Comment by ericlewis** [comment]
@zhongwuzw yeah! The crash was happening because I left some bad debug code behind 😜

**Comment by toddrimes** [comment]
Hello - how can I install the pre-release package that has this fix?

I'm trying
```
react-native init RN060 --version react-native@next
```
but it just keeps  installing v0.57.1

Thank you,
~Todd

**Comment by ericlewis** [comment]
This fix is in 0.59.9. 

Sent via Superhuman iOS ( https://sprh.mn/?vip=ericlewis777@gmail.com )

**Comment by GuoZhiQiang** [comment]
Thanks ！

**Comment by luisfusim** [comment]
How can i get this solution on without update my project, im on 0.56 i code the change directly but im not sure about it (it works), witch could be the best practice (without update) ?
Thanks

**Comment by Base29** [comment]
same as @luisfusim I am too having this issue ... I am on 0.57.0 and don't want to upgrade the RN version ... So what steps I should take to resolve this ... 
Xcode version = 11.0

image

**Comment by tuanduc2904** [comment]
> same as @luisfusim I am too having this issue ... I am on 0.57.0 and don't want to upgrade the RN version ... So what steps I should take to resolve this ...
> Xcode version = 11.0
> 
> image

in your project directory, open file in `myproject/node_modules/react-native/React/Base/RCTModuleMethod.mm`. Then correct as below: [here](https://github.com/facebook/react-native/pull/25146/files#diff-263fc157dfce55895cdc16495b55d190)

**Comment by Base29** [comment]
@tuanduc2904 ... Thanks ... worked like a charm ... :)

**Comment by andreifilip123** [comment]
@tuanduc2904 that will only work until you clean your node_modules. Other ideas ?

**Comment by chengsam** [comment]
Upgrading to 0.59.9 worked for me.

**Comment by rizshivalli** [comment]
> > same as @luisfusim I am too having this issue ... I am on 0.57.0 and don't want to upgrade the RN version ... So what steps I should take to resolve this ...
> > Xcode version = 11.0
> 
> in your project directory, open file in `myproject/node_modules/react-native/React/Base/RCTModuleMethod.mm`. Then correct as below: [here](https://github.com/facebook/react-native/pull/25146/files#diff-263fc157dfce55895cdc16495b55d190)

amazing, This worked for me.

**Comment by russelRajitha** [comment]
> > same as @luisfusim I am too having this issue ... I am on 0.57.0 and don't want to upgrade the RN version ... So what steps I should take to resolve this ...
> > Xcode version = 11.0
> 
> in your project directory, open file in `myproject/node_modules/react-native/React/Base/RCTModuleMethod.mm`. Then correct as below: [here](https://github.com/facebook/react-native/pull/25146/files#diff-263fc157dfce55895cdc16495b55d190)

working perfectly. you saved my day.

**Comment by rewieer** [comment]
> @tuanduc2904 that will only work until you clean your node_modules. Other ideas ?

You can use patch-package to perform the update every time you install packages / update them.

**Comment by tuanduc2904** [comment]
> > @tuanduc2904 that will only work until you clean your node_modules. Other ideas ?
> 
> You can use patch-package to perform the update every time you install packages / update them.

I can't,  do you have a solution?

**Comment by rewieer** [comment]
Well, I just tried, looks like the diff is too heavy for github.

**Comment by tuanduc2904** [comment]
I had to revise it many times when adding new packages to my project

**Comment by eyupcimen** [comment]
> > same as @luisfusim I am too having this issue ... I am on 0.57.0 and don't want to upgrade the RN version ... So what steps I should take to resolve this ...
> > Xcode version = 11.0
> 
> in your project directory, open file in `myproject/node_modules/react-native/React/Base/RCTModuleMethod.mm`. Then correct as below: [here](https://github.com/facebook/react-native/pull/25146/files#diff-263fc157dfce55895cdc16495b55d190)

This working. Thank you

**Comment by GleidsonDaniel** [comment]
Tks @tuanduc2904 , saved my time!

RN version 0.59.10, worked like a charm!

**Comment by 29er** [comment]
why is this  closed. I still get the error ? do I really have to go modify RCTModuleMethod every time in install ????

**Comment by Rambatino** [comment]
Just upgrade react-native package @29er

**Comment by 29er** [comment]
ok thanks @Rambatino ! . I will try 35.0

**Comment by kailash1805a** [comment]
It's working 
https://github.com/facebook/react-native/pull/25146/files#diff-263fc157dfce55895cdc16495b55d190

**Comment by lucky-vishwakarma** [comment]
I also have same issue but these solution does not work for me. I am using
 xcode 11.3
React native 57.8

**Comment by jayasme** [comment]
1) Upgrade to RN 0.59.9 or above in your package.json
2) `rm -rf node_modules/`
3) `yarn`
4) `pod install`
5) Clean in Xcode
6) `yarn start --reset-cache`
7) Build in Xcode

File: React/Base/RCTModuleMethod.mm [hunk 1]
```diff
@@ -91,6 +91,7 @@ static BOOL RCTParseSelectorPart(const char **input, NSMutableString *selector)
 static BOOL RCTParseUnused(const char **input)
 {
   return RCTReadString(input, "__attribute__((unused))") ||
+         RCTReadString(input, "__attribute__((__unused__))") ||
```
Comment (AshishKapoor) [inline React/Base/RCTModuleMethod.mm:94]: Works!

**Review by aasif1297 — COMMENTED** [review]
not working

File: React/Base/RCTModuleMethod.mm (hunk 1 above)
Comment (NamelessLee) [reply React/Base/RCTModuleMethod.mm:94]: Hi, may I know which file this is please? Thanks very much!

**Pull request closed without merging**